azure/resource-group/variables.tf

```diff
@@ -3,14 +3,65 @@ variable "name" {
   type        = string
 }
 
+variable "product" {
+  description = "Name of the project to which the resources belongs."
+  type        = string
+
+  validation {
+    condition     = can(coalesce(var.product))
+    error_message = "The 'project' value is invalid. It must be a non-empty string."
+  }
+}
+
+variable "squad" {
+  description = "Name of the squad to which the resources belongs."
+  type        = string
+
+  validation {
+    condition     = can(coalesce(var.squad))
+    error_message = "The 'squad' value is invalid. It must be a non-empty string."
+  }
+}
+
+variable "country" {
+  description = "Name of the contry to which the resources belongs."
+  type        = string
+
+  validation {
+    condition     = can(coalesce(var.country))
+    error_message = "The 'squad' value is invalid. It must be a non-empty string."
+  }
+}
+
+variable "environment" {
+  description = "The environment in which the resource should be provisioned."
+  type        = string
+
+  validation {
+    condition     = contains(["Dev", "Prod", "Staging", "Test", ], var.environment)
+    error_message = "The 'environment' value is invalid. Valid options are 'Dev', 'Prod','Staging', 'Test'."
+  }
+}
+
 variable "tags" {
-  description = "Azure resource tags map."
+  description = "A mapping of tags which should be assigned to the desired resource."
   type        = map(string)
+  default     = {}
+
+  validation {
+    condition     = alltrue([for tag in var.tags : can(coalesce(tag))])
+    error_message = "At least on tag value from 'tags' is invalid. They must be a non-empty string."
+  }
 }
 
 variable "location" {
   # For a complete list of available Azure regions run at cli:  
   # az account list-locations  --query "[].{displayName:displayName, location:name}" --output table
-  description = "(Required) The Azure Region where the Resource Group should exist. Changing this forces a new Resource Group to be created."
+  description = "The Azure Region where the resource should exist."
   type        = string
+
+  validation {
+    condition     = can(coalesce(var.location))
+    error_message = "The 'location' value is invalid. It must be a non-empty string."
+  }
 }
```

Comment:
I think we should consider this as local since we only support _westeurope_ and no other region.

Reply:
I've added a default value 'westeurope' for the location, to make it flexible in cases we need to experiment with our modules outside the 'westeuope' location. It was just an idea. Please, let me know if you still think that we should add the location as "local".